cmake/compiler/arcmwdt/generic.cmake

```diff
@@ -40,14 +40,15 @@ if(ret)
 else()
   set(ARCMWDT_MIN_REQUIRED_VERS "2022.06")
 #
-# Regular version have format: "T-2022.06"
+# Regular version has format: "T-2022.06"
 # Engineering builds: "ENG-2022.06-001"
+# Check-in build: ENG-2022.12-D1039-C39098348
```

Comment:
I wouldn't go into so many details and if I may agree to have these examples in the commit message, IMHO it's not needed in the body of the change as there might be more variations: think of another custom build where I add "Alexey" **prefix** in the beginning.

What's important is more generalized form we're rectifying with time and as of today it's like: `xxx-YEAR.MM-qwerty` and we're only interested in extraction of "YEAR.MM" part which could be done in even more generic way.

I.e. maybe we even need to drop an assumption of "ENG" or any other letter in the beginning of the version.

Reply:
> I wouldn't go into so many details and if I may agree to have these examples in the commit message, IMHO it's not needed in the body

I'd tend to disagree - it's generally good to know what version strings we've initially targeted the regex for. Of course it would even better to have a tested ... :)